### PR TITLE
refactor(domToReact): refactoring  multiple types

### PR DESCRIPTION
**Pull request by ssi02014**
Release-As: 5.0.2

## What is the motivation for this pull request?

Stricter management of constants and improved types.

## What is the current behavior?

There is no change in behavior.

## What is the new behavior?

There is no change in behavior.

## Checklist:

- [x] [Conventional Commits](https://www.conventionalcommits.org/)
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: src/dom-to-react.ts [hunk 1]
```diff
@@ -37,10 +37,9 @@ export default function domToReact(
   const { cloneElement, createElement, isValidElement } =
     options?.library || React;
 
-  let index = 0;
   const nodesLength = nodes.length;
 
-  for (; index < nodesLength; index++) {
+  for (let index = 0; index < nodesLength; index++) {
```
Comment (ssi02014) [inline src/dom-to-react.ts:42]: How about adding `let index = 0` inside the for statement?

File: src/attributes-to-props.ts [hunk 2]
```diff
@@ -9,8 +9,13 @@ import { PRESERVE_CUSTOM_ATTRIBUTES, setStyleProp } from './utilities';
...
 // https://developer.mozilla.org/docs/Web/HTML/Attributes
-const UNCONTROLLED_COMPONENT_ATTRIBUTES = ['checked', 'value'];
-const UNCONTROLLED_COMPONENT_NAMES = ['input', 'select', 'textarea'];
+const UNCONTROLLED_COMPONENT_ATTRIBUTES = ['checked', 'value'] as const;
+const UNCONTROLLED_COMPONENT_NAMES = ['input', 'select', 'textarea'] as const;
+
+type UncontrolledComponentAttributes =
+  (typeof UNCONTROLLED_COMPONENT_ATTRIBUTES)[number];
+
+type UncontrolledComponentNames = (typeof UNCONTROLLED_COMPONENT_NAMES)[number];
```
[the remark this answers is not in the document]
Comment (remarkablemark) [reply src/attributes-to-props.ts:18]: Looks good

File: src/dom-to-react.ts (hunk 1 above)
Comment (remarkablemark) [reply src/dom-to-react.ts:42]: Works

File: src/index.ts [hunk 3]
```diff
@@ -21,7 +21,7 @@ const domParserOptions = { lowerCaseAttributeNames: false } as const;
 export default function HTMLReactParser(
   html: string,
   options?: HTMLReactParserOptions,
-): JSX.Element | JSX.Element[] | string {
+): ReturnType<typeof domToReact> {
```
[the remark this answers is not in the document]
Comment (remarkablemark) [reply src/index.ts:24]: Makes sense

**Review by remarkablemark — APPROVED** [review]

**Comment by remarkablemark** [comment]
Release:

- https://github.com/remarkablemark/html-react-parser/pull/1134#issuecomment-1785501152

**Comment by ssi02014** [comment]
@remarkablemark Thanks! 🎉

**Pull request merged**